evalml/tests/model_understanding_tests/test_permutation_importance.py

```diff
@@ -0,0 +1,274 @@
+              (LinearPipelineTwoEncoders, {'One Hot Encoder': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                           'One Hot Encoder_2': {'features_to_encode': ['region', 'country']}}),
+              (LinearPipelineWithTextFeatures, {'Drop Columns Transformer': {'columns': ['datetime']},
+                                                'Text Featurization Component': {'text_columns': ['provider']}}),
+              (LinearPipelineWithTextFeatures, {'Drop Columns Transformer': {'columns': ['datetime']}}),
+              (LinearPipelineWithDoubling, {'Select Columns Transformer': {'columns': ['amount']}}),
+              (LinearPipelineWithDoubling, {'Select Columns Transformer': {'columns': ['amount']},
+                                            'DoubleColumns': {'drop_old_columns': False}}),
+              (DagTwoEncoders, {'SelectNumeric': {'columns': ['card_id', 'store_id', 'datetime', 'amount', 'customer_present', 'lat', 'lng']},
+                                'SelectCategorical1': {'columns': ['currency', 'expiration_date', 'provider']},
+                                'SelectCategorical2': {'columns': ['region', 'country']},
+                                'OHE_1': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                'OHE_2': {'features_to_encode': ['region', 'country']}}),
+              (DagReuseFeatures, {'SelectNumeric': {'columns': ['card_id', 'store_id', 'datetime', 'amount', 'customer_present', 'lat', 'lng']},
+                                  'SelectCategorical1': {'columns': ['currency', 'expiration_date', 'provider']},
+                                  'SelectCategorical2': {'columns': ['region', 'country']},
+                                  'OHE_1': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                  'OHE_2': {'features_to_encode': ['region', 'country']},
+                                  'DT': {'encode_as_categories': True}}),
+              (LinearPipelineWithTargetEncoderAndOHE, {'One Hot Encoder': {'features_to_encode': ['currency', 'expiration_date', 'provider']},
+                                                       'Target Encoder': {'cols': ['region', 'country']}}),
+              (LinearPipelineCreateFeatureThenDropIt, {'Select Columns Transformer': {'columns': ['amount']},
+                                                       'DoubleColumns': {'drop_old_columns': False},
+                                                       'Drop Columns Transformer': {'columns': ['amount_doubled']}})
+              ]
+
+
+@pytest.mark.parametrize('pipeline_class, parameters', test_cases)
+@patch('evalml.pipelines.PipelineBase._supports_fast_permutation_importance', new_callable=PropertyMock)
+def test_fast_permutation_importance_matches_sklearn_output(mock_supports_fast_importance, pipeline_class, parameters,
```

Comment:
I think it's important to test that our optimization calculates the values correctly. Mocking doesn't really make sense here but since I'm only using 100 points and `n_jobs=1` I think it's ok.